omnigibson/utils/usd_utils.py

```diff
@@ -325,18 +469,46 @@ def in_contact(cls, prim_paths_a, prim_paths_b):
         """
         # Check if the contact tuple already exists in the cache; if so, return the value
         key = (tuple(prim_paths_a), tuple(prim_paths_b))
-        if key not in cls._CONTACT_CACHE:
+        if key not in self._CONTACT_CACHE:
             # In contact if any of the matrix values representing the interaction between the two groups is non-zero
-            cls._CONTACT_CACHE[key] = np.any(cls.get_impulses(prim_paths_a=prim_paths_a, prim_paths_b=prim_paths_b))
-        return cls._CONTACT_CACHE[key]
+            self._CONTACT_CACHE[key] = np.any(self.get_impulses(prim_paths_a=prim_paths_a, prim_paths_b=prim_paths_b))
+        return self._CONTACT_CACHE[key]
 
-    @classmethod
-    def clear(cls):
+    def clear(self):
         """
         Clears the internal contact matrix and cache
         """
-        cls._CONTACT_MATRIX = None
-        cls._CONTACT_CACHE = dict()
+        self._CONTACT_MATRIX = dict()
+        self._CONTACT_DATA = dict()
+        self._CONTACT_CACHE = dict()
+
+
+# Instantiate the RigidContactAPI
+RigidContactAPI = RigidContactAPIImpl()
+
+
+class GripperRigidContactAPIImpl(RigidContactAPIImpl):
+    @classmethod
+    def get_column_filters(cls):
+        from omnigibson.robots.manipulation_robot import ManipulationRobot
+
+        filters = dict()
+        for scene_idx, scene in enumerate(og.sim.scenes):
+            filters[scene_idx] = []
+            for robot in scene.robots:
+                if isinstance(robot, ManipulationRobot):
+                    filters[scene_idx].extend(link.prim_path for links in robot.finger_links.values() for link in links)
+
+        return filters
+
+    @classmethod
+    def get_max_contact_data_count(cls):
+        # 2x per finger link, to be safe.
+        return len(cls.get_column_filters()[0]) * 2
```

Comment:
hmm seems hacky, what if a robot has more than 2 fingers?

Reply:
2 here is not the finger count, it's the number of items we will record contacts with, **per finger**. e.g. it's N such that if the finger is touching more than N items at once, only the first N are recorded. This number should very rarely go above 2.

Reply:
TODO: add comment